docs/advanced/data_types.mdx

```diff
@@ -54,6 +54,34 @@ To add any code documentation website as a loader, use the data_type as `docs_si
 app.add("docs_site", "https://docs.embedchain.ai/")
 ```
 
+### Gmail
+
+To add mails from your gmail inbox, use the data_type as `gmail`.
+
+The second argument is a search query. You can test queries in your gmail web client.
+For all mails, use `*` (behaviour with large inboxes has not been tested, use at your own risk).
+
+To use this you need `credentials.json` in your current directory. Follow these steps to get the credentials
+
+1. Go to the [Google Cloud Console](https://console.cloud.google.com/apis/credentials).
+2. Create a project if you don#t have one already.
+3. Create an `OAuth Consent Screen` in the project. Don't be alarmed when you have to choose `external`.
+4. Make sure the consent screen is published.
+5. Enable the [Gmail API](https://console.cloud.google.com/apis/api/gmail.googleapis.com)
+6. Create credentials from the `Credentials` tab.
+7. Select the type `OAuth Client ID`.
+8. Choose the application type `Web application`. As a name you can choose `embedchain`.
+9. Add an authorized redirect URI for `http://localhost:8080/`.
+10. You can leave everything else at default, finish the creation.
+11. When you are done, a modal opens where you can download the details in `json` format.
+12. Put the `.json` file in your current directory and rename it to `credentials.json`
+
+Eg:
+
+```python
+app.add("gmail", "*")
```

Comment:
I followed the instruction, but I am getting following error from gmail.

```
Access blocked: This app’s request is invalid

You can’t sign in because this app sent an invalid request. You can try again later, or contact the developer about this issue. [Learn more about this error](https://developers.google.com/identity/protocols/oauth2/web-server#authorization-errors-redirect-uri-mismatch)
If you are a developer of this app, see [error details](https://accounts.google.com/).
Error 400: redirect_uri_mismatch
```

I entered the redirect uri as localhost:8080, but I think we will need to run the server. Right now the server is not running.

Reply:
No I never had to run a server specifically, I guess it starts a server for authentication.

Reply:
So when I run the code, it opens the browser with `https://accounts.google.com/o/oauth2/auth/oauthchooseaccount`

Reply:
Choosing an account there redirects to port localhost:8080, which displays: `The authentication flow has completed. You may close this window.`

Reply:
I can try with again with another account if you want.